torch/csrc/Dtype.cpp

```diff
@@ -29,13 +29,29 @@ PyObject *THPDtype_is_floating_point(THPDtype *self)
   }
 }
 
+PyObject *THPDtype_reduce(THPDtype *self)
+{
+  /*
+  * For singletons, a string is returned. The string should be interpreted
+  * as the name of a global variable.
+  */
+  return THPUtils_packString(self->name);
+}
+
 typedef PyObject *(*getter)(PyObject *, void *);
 
 static struct PyGetSetDef THPDtype_properties[] = {
   {"is_floating_point", (getter)THPDtype_is_floating_point, nullptr, nullptr, nullptr},
   {nullptr}
 };
 
+static PyMethodDef THPDtype_methods[] = {
+  {"__reduce__", (PyCFunction)THPDtype_reduce, METH_NOARGS,
+   "define behavior in pickling"
```

Comment:
You can just pass a NULL pointer as docstring. It's not going to be very useful here.